api/src/main/java/openconsensus/OpenConsensus.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019, OpenConsensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package openconsensus;
+
+import java.util.concurrent.atomic.AtomicReference;
+import openconsensus.trace.NoopTrace;
+import openconsensus.trace.Trace;
+import openconsensus.trace.Tracer;
+
+/**
+ * Registry provides a global access to observability objects. An implementation is responsible for
+ * setting the Registry.
+ */
+public class OpenConsensus {
+
+  private static final OpenConsensus INSTANCE = new OpenConsensus();
+
+  private final AtomicReference<Tracer> tracer = new AtomicReference<>(Trace.getTracer());
+
+  /** Reset to the initial state. */
+  public static void reset() {
+    INSTANCE.tracer.set(null);
+  }
+
+  /**
+   * Returns an instance of a {@link Tracer}.
+   *
+   * @return registered tracer or {@link NoopTrace} singleton via {@link Trace#getTracer()}.
+   */
+  public static Tracer getTracer() {
+    return INSTANCE.tracer.get();
+  }
+
+  /**
+   * Sets a {@link Tracer} to the registry.
+   *
+   * @param newInstance tracer to be registered
+   * @exception IllegalStateException if a tracer is already registered
+   */
+  public static void setTracer(Tracer newInstance) {
```

Comment:
`setTracer` disable the ability to cache the result of `getTracer`, is this the behavior we want?

In an ideal world I would like to write something like `Logger` in my class:
`private static final Tracer tracer = OpenConsensus.getTracer();`

Do we want to achieve this?